BGL/include/CGAL/boost/graph/named_params_helper.h

```diff
@@ -175,37 +177,105 @@ namespace CGAL {
     > ::type  type;
   };
 
-  template<typename PolygonMesh, typename NamedParameters>
-  class GetFaceIndexMap
-  {
-    typedef typename property_map_selector<PolygonMesh, boost::face_index_t>::type DefaultMap;
-    typedef typename property_map_selector<PolygonMesh, boost::face_index_t>::const_type DefaultMap_const;
-  public:
-    typedef typename internal_np::Lookup_named_param_def <
-    internal_np::face_index_t,
-    NamedParameters,
-    DefaultMap
-    > ::type  type;
-    typedef typename internal_np::Lookup_named_param_def <
-      internal_np::face_index_t,
-      NamedParameters,
-      DefaultMap_const
-      > ::type  const_type;
-    typedef typename boost::is_same<type, DefaultMap>::type Is_internal_map;
-    typedef typename boost::is_same<const_type, DefaultMap_const>::type Is_internal_map_const;
-  };
-
-  template<typename PolygonMesh, typename NamedParameters>
-  class GetVertexIndexMap
-  {
-    typedef typename property_map_selector<PolygonMesh, boost::vertex_index_t>::type DefaultMap;
-  public:
-    typedef typename internal_np::Lookup_named_param_def <
-    internal_np::vertex_index_t,
-    NamedParameters,
-    DefaultMap
-    > ::type  type;
-  };
+// Define the following structs:
+//
+// GetInitializedVertexIndexMap
+// GetInitializedHalfedgeIndexMap
+// GetInitializedEdgeIndexMap
+// GetInitializedFaceIndexMap
+
+#define CGAL_DEF_GET_INDEX_TYPE(CTYPE, TYPE)                                                       \
+template <typename Graph,                                                                          \
+          typename NamedParameters =                                                               \
+            CGAL::Named_function_parameters<bool, CGAL::internal_np::all_default_t> >              \
+struct GetInitialized##CTYPE##IndexMap                                                             \
+  : public BGL::internal::GetInitializedIndexMap<internal_np::TYPE##_index_t,                      \
+                                                 boost::TYPE##_index_t,                            \
+                                                 CGAL::dynamic_##TYPE##_property_t<int>,           \
```

Comment:
any reason not to use `unsigned int`?

Reply:
Should it actually even be `vertices_size_type`, `halfedges_size_type`, ...?

Reply:
Agreed